channel.js

```diff
@@ -282,6 +286,13 @@ function TChannel(options) {
         }
         self.sanityTimer = self.timers.setTimeout(doSanitySweep, SANITY_PERIOD);
     }
+
+    this.enableMaxRetryRatio = this.options.enableMaxRetryRatio || DEFAULT_ENABLE_MAX_RETRY_RATIO;
+    if (this.enableMaxRetryRatio) {
+        this.maxRetryRatio = this.options.maxRetryRatio || DEFAULT_MAX_RETRY_RATIO;
```

Comment:
we should not conditionally assign fields in the constructor as that de-optimizes the code.

Please pre-assign them outside the if statement

```
this.maxRetryRation = DEFAULT_MAX_RETRY_RATIO;
this.retryRatioTracker = null;
if (this.enableMaxRetryRatio) {
  ...
}
```